--- fisheyectl.sh	2015-12-30 23:00:39.323145257 +0000
+++ fisheyectl.sh-with-pid	2015-12-30 23:12:47.662769775 +0000
@@ -54,7 +54,20 @@
   mkdir -p $FISHEYE_INST/var/log
   CMD="$FISHEYE_CMD $@ $FISHEYE_ARGS"
   echo "Starting FishEye/Crucible... Output redirected to $FISHEYE_INST/var/log/fisheye.out"
+  CUR_NUM_LINES=$( (wc -l $FISHEYE_INST/var/log/fisheye.out 2> /dev/null || echo 0) | awk '{print $1}')
   nohup sh -c "exec $CMD" >> $FISHEYE_INST/var/log/fisheye.out 2>&1 &
+  PID=$!
+  timeout=600
+  while [ "$timeout" -ge 0 -a -z "$(tail -n+$CUR_NUM_LINES $FISHEYE_INST/var/log/fisheye.out|grep 'Server started')" ] ; do
+    timeout=$((timeout-1))
+    sleep .1s
+  done
+  if [ "$timeout" -ge 0 ]; then
+    echo $PID > $FISHEYE_INST/fecru.pid
+  else
+    echo "Failed to start FishEye/Crucible within 60s... Terminating"
+    exit 1
+  fi
 else
   exec $FISHEYE_CMD $@ $FISHEYE_ARGS
 fi
